server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalDateHistogram.java

```diff
@@ -352,21 +353,56 @@ protected Bucket reduceBucket(List<Bucket> buckets, ReduceContext context) {
     }
 
     private void addEmptyBuckets(List<Bucket> list, ReduceContext reduceContext) {
-        Bucket lastBucket = null;
-        LongBounds bounds = emptyBucketInfo.bounds;
+        /*
+         * Make sure we have space for the empty buckets we're going to add by
+         * counting all of the empties we plan to add and firing them into
+         * consumeBucketsAndMaybeBreak.
+         *
+         * We don't count all of the buckets we plan to allocate and then report
+         * them all at once because we you can configure the date_histogram to
+         * create an astounding number of buckets. It'd take a while to count
+         * that high only to abort. So we report every couple thousand buckets.
+         * It's be simpler to report every single bucket we plan to allocate
+         * one at a time but that'd cause needless overhead on the circuit
+         * breakers. Counting a couple thousand buckets is plenty fast to fail
+         * this quickly in pathological cases and plenty large to keep the
+         * overhead minimal.
+         */
+        int reportEmptyEvery = 10000;
```

Comment:
Should this be a constant?

Reply:
Probably. I stuck it here so the comment above it could describe why it has the value it does. I'm not sure the right way to do that if its a constant without making it harder to read.

Reply:
Can you at least make it final? My first reaction was "Why and where is he changing it?".

Reply:
Sure!